### PR TITLE
fix(linter): unexpected unwrap panic

### PR DESCRIPTION
**Pull request by msdlisper**
While reviewing the code, I noticed that a panic will occur if the object does not have the 'ignoreNonDOM' key, which is unexpected. And I have moved out the code `no_focus.set_option(true);` to make the function cleaner.

### CODE REVIEW

File: crates/oxc_linter/src/rules/jsx_a11y/no_autofocus.rs [hunk 1]
```diff
@@ -73,16 +73,16 @@ impl Rule for NoAutofocus {
     fn from_configuration(value: serde_json::Value) -> Self {
         let mut no_focus = Self::default();
 
-        let _ = value.as_array().unwrap().iter().find(|v| {
+        if value.as_array().unwrap().iter().any(|v| {
```
Comment (camc314) [inline crates/oxc_linter/src/rules/jsx_a11y/no_autofocus.rs:76]: Won't this also panic if it's now an array?

Comment (msdlisper) [reply crates/oxc_linter/src/rules/jsx_a11y/no_autofocus.rs:76]: Ooh, right, I noticed that the config can only a number like this :
```
{
  "rules": {
    "jsx-a11y/rule-name": 2
  }
}
```
or just an object:
```
{
  "rules": {
    "jsx-a11y/rule-name": {
            "ignoreNonDOM": true
        }
  }
}
```

Comment (msdlisper) [reply crates/oxc_linter/src/rules/jsx_a11y/no_autofocus.rs:76]: Maybe a public method should be distilled here 🤔

File: crates/oxc_linter/src/utils/config.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * or
+ *
+ * {
+ *  "rules": {
+ *     "jsx-a11y/rule-name": [{
+ *         "ignoreNonDOM": true
+ *       }
+ *    }]
+ * }
+ */
```
Comment (camc314) [inline crates/oxc_linter/src/utils/config.rs:28]: i don't think we should have this file - instead do it direcly inside `from_configuration`

I also think that we always do the following:

if the config looks like 
```json
{ 
    "<RULE NAME>": [2, { "config": {} }],
}
```

we just pass in `[{config}]` to the `from_configuration` call, ignoreing the first element.

based on the eslint docs, if a rule configuration is passed, it _has_ to look like `<RULE NAME>: [<SEVERITY>, config]`

**Review by camc314 — APPROVED** [review]
Thank you!

**Pull request merged**